src/Lean/Elab/Tactic/BVDecide/Frontend/Normalize/Structures.lean

```diff
@@ -18,15 +18,47 @@ it is a non recursive structure and at least one of the following conditions hol
 - it contains something of type `BitVec`/`UIntX`/`Bool`
 - it is parametrized by an interesting type
 - it contains another interesting type
-Afterwards we also apply relevant `injEq` theorems to support at least equality for these types out
-of the box.
+Afterwards we also:
+- apply relevant `injEq` theorems to support at least equality for these types out of the box.
+- push projections of relevant types inside of `ite` and `cond`.
 -/
 
 namespace Lean.Elab.Tactic.BVDecide
 namespace Frontend.Normalize
 
 open Lean.Meta
 
+
+def applyIteSimproc : Simp.Simproc := fun e => do
+  let proj := e.getAppFn
+  let args := e.getAppArgs
+  if h : args.size ≠ 0 then
```

Comment:
Do you know `withApp`? Quite nice for applications like this:
```suggestion
def applyIteSimproc : Simp.Simproc := fun e => e.withApp fun proj args => do
  if h : args.size ≠ 0 then
```
(Very minor suggestions, of course)